src/Cool/DidYouKnow/didYouKnowModal.cpp

```diff
@@ -0,0 +1,98 @@
+#include "didYouKnowModal.hpp"
+#include <imgui.h>
+#include <vcruntime.h>
+#include <string>
+
+void DidYouKnowModal::open()
+{
+    auto difference = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()) - std::chrono::system_clock::to_time_t(_current_timestamp);
+
+    size_t time_to_wait = 1 * 60 * 60; // 1 hour
+
+    if (!_has_been_opened && difference > time_to_wait)
+    {
+        _has_been_opened = true;
+        ImGui::OpenPopup(_id.c_str());
+        _current_timestamp = std::chrono::system_clock::now();
```

Comment:
You should use `std::chrono::steady_clock()` all over the place. `std::chrono::system_clock` is affected by daylight saving time, changing timezone, leap seconds, etc. which we do not want.